guides/05_chatbots/03_agents-and-tool-usage.md

```diff
@@ -166,4 +167,139 @@ demo.launch()
 That's it! See our finished langchain demo [here](https://huggingface.co/spaces/gradio/langchain-agent).
 
 
+## Building with Visible Thinking LLMs
+
+
+The Gradio Chatbot can natively display intermediate thoughts of a _thinking_ LLM. This makes it perfect for creating UIs that show how an AI model "thinks" while generating responses. Below guide will show you how to build a chatbot that displays Gemini AI's thought process in real-time.
+
+
+### A real example using Gemini2.0 Flash Thinking API
+
+Let's create a complete chatbot that shows its thoughts and responses in real-time. We'll use Google's Gemini API for accessing Gemini2.0 Flash Thinking LLM and Gradio for the UI.
+
+We'll begin with imports and setting up the gemini client. Note that you'll need to acquire a Google Gemini API key first -
```

Comment:
Perhaps you can link to where one can obtain a Gemini API key
```suggestion
We'll begin with imports and setting up the gemini client. Note that you'll need to acquire a Google Gemini API key first -
```

Reply:
Good idea! Will add a hyperlink.